dipy/tracking/fast_tracking.pyx

```diff
@@ -0,0 +1,235 @@
+        raise MemoryError("Memory allocation failed")
+
+    # srand/rand don't play well with multi-threading
+    if params.random_seed > 0 and not nbr_threads == 1:
+        raise ValueError("random_seed > 0 do not work with nbr_threads != 1.")
+
+    seed_start = 0
+    seed_end = seed_start + _plen
+    while seed_start < _len:
+        generate_tractogram_c(seed_positions[seed_start:seed_end],
+                              seed_directions[seed_start:seed_end],
+                              nbr_threads, sc, params,
+                              pmf_gen,
+                              streamlines_arr, length_arr, status_arr)
+        seed_start += _plen
+        seed_end += _plen
+        streamlines = []
+        try:
+            for i in range(_len):
+                if length_arr[i] > 1:
+                    s = np.asarray(<cnp.float_t[:length_arr[i]*3]> streamlines_arr[i])
+                    streamlines.append(s.copy().reshape((-1,3)))
+                    if streamlines_arr[i] == NULL:
+                        continue
+                    free(streamlines_arr[i])
+        finally:
+            free(streamlines_arr)
+            free(length_arr)
+            free(status_arr)
+
+        for s in streamlines:
+            yield s
+
+
+cdef int generate_tractogram_c(double[:,::1] seed_positions,
+                               double[:,::1] seed_directions,
+                               int nbr_threads,
+                               StoppingCriterion sc,
+                               TrackerParameters params,
+                               PmfGen pmf_gen,
+                               double** streamlines,
+                               int* lengths,
+                               int* status):
+    cdef:
+        cnp.npy_intp _len=seed_positions.shape[0]
+        cnp.npy_intp i, j, k
+
+    if nbr_threads<= 0:
+        nbr_threads = 0
+    for i in prange(_len, nogil=True, num_threads=nbr_threads):
+        stream = <double*> malloc((params.max_len * 3 * 2 + 1) * sizeof(double))
+        stream_idx = <int*> malloc(2 * sizeof(int))
+        status[i] = generate_local_streamline(&seed_positions[i][0],
+                                              &seed_directions[i][0],
+                                              stream,
+                                              stream_idx,
+                                              sc,
+                                              params,
+                                              pmf_gen)
+
+        # copy the streamlines points from the buffer to a 1d vector of the streamline length
+        lengths[i] = stream_idx[1] - stream_idx[0]
+        if lengths[i] > 1:
+            streamlines[i] = <double*> malloc(lengths[i] * 3 * sizeof(double))
+            memcpy(&streamlines[i][0], &stream[stream_idx[0] * 3], lengths[i] * 3 * sizeof(double))
+        free(stream)
+        free(stream_idx)
+
+    return 0
```

Comment:
Looks like this is the only possible return value. Maybe define a constant called e.g. ``TRACKING_SUCCESS` and assign it a 0 value and use it?

Reply:
I change the return from int to void.